flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java

```diff
@@ -429,4 +495,46 @@ protected AsynchronousPostStopEndpoint(RpcService rpcService, CompletableFuture<
 			return postStopFuture;
 		}
 	}
+
+	// -------------------------------------------------------------------------
+
+	interface OversizedResponseMsgRpcGateway extends RpcGateway {
+		CompletableFuture<Byte[]> calculate();
+	}
+
+	static class OversizedResponseRpcEndpoint extends TestRpcEndpoint implements OversizedResponseMsgRpcGateway {
+
+		private volatile Byte[] bytes = new Byte[1024 * 10];
+
+		public OversizedResponseRpcEndpoint(RpcService rpcService) {
+			super(rpcService);
+		}
+
+		@Override
+		public CompletableFuture<Byte[]> calculate() {
+			return CompletableFuture.completedFuture(bytes);
+		}
+
+		public Byte[] getBytes() {
+			return bytes;
+		}
+
+		public void setBytes(Byte[] bytes) {
+			this.bytes = bytes;
+		}
+	}
+
+	static class TestingRemoteRpcService extends TestingRpcService {
```

Comment:
No need for this `RpcService` if we properly simulate remote communication by using a second `RpcService`.